CMakeLists.txt

```diff
@@ -95,10 +98,30 @@ zephyr_compile_options(
   -Wformat
   -Wformat-security
   -Wno-format-zero-length
+  -imacros ${AUTOCONF_H}
+)
+
+
+# FIXME
+if(CONFIG_ARCH_POSIX)
+zephyr_compile_options(
+  -Wno-undef
+  -Wno-implicit-function-declaration
+  )
+else()
+zephyr_compile_options(
   -Wno-main
   -ffreestanding
-  -include ${AUTOCONF_H}
+  )
+endif()
+
+if(CONFIG_ARCH_POSIX)
```

Comment:
I assume we can get rid of this "if POSIX" since I assume COVERAGE already
depends on POSIX through Kconfig.

Reply:
=> @nashif